agent/consul/leader.go

```diff
@@ -355,6 +381,413 @@ func (s *Server) initializeACL() error {
 	return nil
 }
 
+// initializeACLs is used to setup the ACLs if we are the leader
+// and need to do this.
+func (s *Server) initializeACLs(upgrade bool) error {
+	if !s.ACLsEnabled() {
+		return nil
+	}
+
+	// Purge the cache, since it could've changed while we were not the
+	// leader.
+	s.acls.cache.Purge()
+
+	if s.InACLDatacenter() {
+		if s.UseLegacyACLs() && !upgrade {
+			s.logger.Printf("[INFO] acl: initializing legacy acls")
+			return s.initializeLegacyACL()
+		}
+
+		s.logger.Printf("[INFO] acl: initializing new acls")
+
+		// Create the builtin global-management policy
+		_, policy, err := s.fsm.State().ACLPolicyGetByID(nil, structs.ACLPolicyGlobalManagementID)
+		if err != nil {
+			return fmt.Errorf("failed to get the builtin global-management policy")
+		}
+		if policy == nil {
+			policy := structs.ACLPolicy{
+				ID:          structs.ACLPolicyGlobalManagementID,
+				Name:        "global-management",
```

Comment:
Do we protect against this policy ever being removed by mistake? What happens if that's possible? I have flashbacks of users who remove the default MySQL admin user and then can't ever get back into their DB...

Reply:
Yes we do, It is undeleteable and the rules are not allowed to be changed. The only thing you can do is rename it as everywhere internally references it by the reserved/well known UUID.